customer-service/src/main/scala/Server.scala

```diff
@@ -1,39 +1,47 @@
 import cats.effect._
 import cats.implicits.toBifunctorOps
-import domain.CustomerService
+import domain.{CustomerService, HttpClient}
 import gateway.CustomerRoutes.customerRoutes
 import infrastructure.repositories.InMemoryCustomerRepository
 import infrastructure.repositories.config.ServerConf
-import org.http4s.HttpApp
 import org.http4s.blaze.server.BlazeServerBuilder
-import org.http4s.server.Router
 import org.http4s.implicits._
+import org.http4s.server.Router
 import pureconfig.ConfigSource
 import pureconfig.generic.auto._
 
 object Server extends IOApp {
 
-  private val customerRepo = new InMemoryCustomerRepository[IO]
-  val customerService      = new CustomerService[IO](customerRepo)
-
-  private val httpApp: HttpApp[IO] = Router(
-    "/api" -> customerRoutes[IO](customerService)
-  ).orNotFound
-
   private def loadConfig: IO[ServerConf] =
     IO.fromEither(
       ConfigSource.default.load[ServerConf].leftMap(error => new RuntimeException(s"Failed to load config $error"))
     )
 
-  override def run(args: List[String]): IO[ExitCode] =
-    loadConfig.flatMap { serverConfig =>
-      BlazeServerBuilder[IO]
+  private def buildCustomerService: CustomerService[IO] = {
```

Comment:
that is very correct, but place is wrong - it should be in infrastructure
```
CustomerModule {
   def customerInMemoryRepository(map): IO[CustomerRepository] = for {
      create ref(map) here
   } yield inMemoryRepo

   def customerService(customerRepository, customerDetailsService): IO[CustomerService]
}
```

and in different file
```
CustomerDetailsModule {
  // same pattern
}
```